diff --git a/lib/client/composables/useDocsearchShim.js b/lib/client/composables/useDocsearchShim.js
index 7ee04844efd953af77719e87bc8003adcb2d3fbd..574bd234567b49c2bdffe8a023aee8388e03032b 100644
--- a/lib/client/composables/useDocsearchShim.js
+++ b/lib/client/composables/useDocsearchShim.js
@@ -26,7 +26,7 @@ export const useDocsearchShim = () => {
                         return;
                     }
                     event.preventDefault();
-                    router.push(resolveRoutePathFromUrl(hit.url, __VUEPRESS_BASE__));
+                    router.push(resolveRoutePathFromUrl(hit.url, __DOCSEARCH_BASE__));
                 },
                 children,
             },
@@ -36,7 +36,7 @@ export const useDocsearchShim = () => {
         navigator: {
             // when pressing Enter without metaKey
             navigate: ({ itemUrl }) => {
-                router.push(resolveRoutePathFromUrl(itemUrl, __VUEPRESS_BASE__));
+                router.push(resolveRoutePathFromUrl(itemUrl, __DOCSEARCH_BASE__));
             },
         },
         // add search debounce
diff --git a/lib/node/docsearchPlugin.d.ts b/lib/node/docsearchPlugin.d.ts
index c03da98424d08a05ccf250ba6471d234b56d76b4..da82395ec6ff07170093de2ac0da9244052d036a 100644
--- a/lib/node/docsearchPlugin.d.ts
+++ b/lib/node/docsearchPlugin.d.ts
@@ -2,5 +2,6 @@ import type { Plugin } from '@vuepress/core';
 import type { DocsearchOptions } from '../shared/index.js';
 export interface DocsearchPluginOptions extends DocsearchOptions {
     injectStyles?: boolean;
+    indexBase?: string;
 }
 export declare const docsearchPlugin: ({ injectStyles, ...options }: DocsearchPluginOptions) => Plugin;
diff --git a/lib/node/docsearchPlugin.js b/lib/node/docsearchPlugin.js
index 5f226a7e69b4bbd92e03f96c9d089986c3315985..2b118cb7892d7de98c99c5c84dcb7d8a823d7ba0 100644
--- a/lib/node/docsearchPlugin.js
+++ b/lib/node/docsearchPlugin.js
@@ -1,10 +1,11 @@
 import { getDirname, path } from '@vuepress/utils';
 const __dirname = getDirname(import.meta.url);
-export const docsearchPlugin = ({ injectStyles = true, ...options }) => ({
+export const docsearchPlugin = ({ injectStyles = true, indexBase = "", ...options }) => ({
     name: '@vuepress/plugin-docsearch',
     clientConfigFile: path.resolve(__dirname, '../client/config.js'),
-    define: {
+    define: (app) => ({
+        __DOCSEARCH_BASE__: indexBase || app.options.base,
         __DOCSEARCH_OPTIONS__: options,
         __DOCSEARCH_INJECT_STYLES__: injectStyles,
-    },
+    }),
 });