diff --git a/dist/index.js b/dist/index.js
index c0fab35fd911bd663965790be188cfd578dedf0e..30b88e48d322145df6996ba3d0e655989814e392 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -120,14 +120,7 @@ import '@vuepress/client/app'
             ...isServer ? {
               // also add hash to ssr entry file, so that users could build multiple sites in a single process
               entryFileNames: `[name].[hash].mjs`
-            } : {
-              manualChunks(id) {
-                if (id.includes("plugin-vue:export-helper") || /node_modules\/@vuepress\/shared\//.test(id) || /node_modules\/vue(-router)?\//.test(id)) {
-                  return "framework";
-                }
-                return void 0;
-              }
-            }
+            } : {}
           },
           preserveEntrySignatures: "allow-extension"
         },