diff --git a/dist/index.js b/dist/index.js
index 76f5ecd69f1df73c1926b988cf917088f7e14c98..f169f3cc848ac1669ac35000b8bd43f8bd655cb6 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -370,15 +370,16 @@ const watchPageFiles = (app) => {
 	const cacheDir = app.dir.cache();
 	const ignoreMatcher = picomatch(ignorePatterns, { cwd: sourceDir });
 	const pageMatcher = picomatch(pagePatterns, { cwd: sourceDir });
-	const pagesWatcher = chokidar.watch(".", {
-		cwd: sourceDir,
-		ignored: (filepath, stats) => {
-			if (ignoreMatcher(filepath)) return true;
-			if (stats?.isDirectory()) return filepath === tempDir || filepath === cacheDir;
-			return !!stats?.isFile() && !pageMatcher(path.relative(sourceDir, filepath));
-		},
-		ignoreInitial: true
-	});
+  const pagesWatcher = chokidar.watch('.', {
+    cwd: sourceDir,
+    ignored: (filepath, stats) => {
+      const relative = path.relative(sourceDir, filepath)
+      if (ignoreMatcher(relative)) return true
+      if (filepath === tempDir || filepath === cacheDir) return true
+      return !!stats?.isFile() && !pageMatcher(relative)
+    },
+    ignoreInitial: true,
+  })
 	pagesWatcher.on("add", async (filePathRelative) => {
 		logger.info(`page ${colors.magenta(filePathRelative)} is created`);
 		const page = await handlePageAdd(app, app.dir.source(filePathRelative));
