diff --git a/.vscode/settings.json b/.vscode/settings.json
new file mode 100644
index 0000000000000000000000000000000000000000..7c2feb7ef4731d430facfcafcbac59e3ea861849
--- /dev/null
+++ b/.vscode/settings.json
@@ -0,0 +1,3 @@
+{
+  "editor.formatOnSave": false
+}
diff --git a/auto/auto.mts b/auto/auto.mts
index f0bc380548f97ae2b5d3ae4c324518a47dd6eeaf..86103fd7fe5141990d521b99f2c63473186087fe 100644
--- a/auto/auto.mts
+++ b/auto/auto.mts
@@ -1,4 +1,4 @@
-import { Chart } from '../types/index.esm';
+import { Chart } from '../types/index.esm.js';
 
-export * from '../types/index.esm';
+export * from '../types/index.esm.js';
 export default Chart;
diff --git a/types/adapters.d.ts b/types/adapters.d.ts
index cae40966ba06989764a5481b0ebc9af9b7439af2..b7be2c29ce850ab5a79ae8dd68ec7beb8753c442 100644
--- a/types/adapters.d.ts
+++ b/types/adapters.d.ts
@@ -1,4 +1,4 @@
-import type { ChartOptions } from './index.esm';
+import type { ChartOptions } from "./index.esm.js";
 
 export type TimeUnit = 'millisecond' | 'second' | 'minute' | 'hour' | 'day' | 'week' | 'month' | 'quarter' | 'year';
 
diff --git a/types/animation.d.ts b/types/animation.d.ts
index b83204124f86972541c0653536b6e1e7e9f1e9d7..a45ae758eab8cee64ec7a0a00af29d2981e117e1 100644
--- a/types/animation.d.ts
+++ b/types/animation.d.ts
@@ -1,5 +1,5 @@
-import { Chart } from './index.esm';
-import { AnyObject } from './basic';
+import { Chart } from "./index.esm.js";
+import { AnyObject } from "./basic.js";
 
 export class Animation {
   constructor(cfg: AnyObject, target: AnyObject, prop: string, to?: unknown);
diff --git a/types/element.d.ts b/types/element.d.ts
index 3b9359b3a1b6abf4d54ec7fbadd066c437de2f60..fa9f1f82c1edb6e1a61113873b88a08b19494a99 100644
--- a/types/element.d.ts
+++ b/types/element.d.ts
@@ -1,5 +1,5 @@
-import { AnyObject } from './basic';
-import { Point } from './geometric';
+import { AnyObject } from './basic.js';
+import { Point } from './geometric.js';
 
 export interface Element<T = AnyObject, O = AnyObject> {
   readonly x: number;
diff --git a/types/index.esm.d.ts b/types/index.esm.d.ts
index 4284e2d1019e7a69af76cad7e2e6b5a37bba5464..b37f62e82b45fe164ea64361d8cad4a05abf4332 100644
--- a/types/index.esm.d.ts
+++ b/types/index.esm.d.ts
@@ -1,19 +1,24 @@
-import { DeepPartial, DistributiveArray, UnionToIntersection } from './utils';
-
-import { TimeUnit } from './adapters';
-import { AnimationEvent } from './animation';
-import { AnyObject, EmptyObject } from './basic';
-import { Color } from './color';
-import { Element } from './element';
-import { ChartArea, Point } from './geometric';
-import { LayoutItem, LayoutPosition } from './layout';
-
-export { DateAdapter, TimeUnit, _adapters } from './adapters';
-export { Animation, Animations, Animator, AnimationEvent } from './animation';
-export { Color } from './color';
-export { Element } from './element';
-export { ChartArea, Point } from './geometric';
-export { LayoutItem, LayoutPosition } from './layout';
+import { DeepPartial, DistributiveArray, UnionToIntersection } from './utils.js';
+
+import { TimeUnit } from "./adapters.js";
+import { AnimationEvent } from "./animation.js";
+import { AnyObject, EmptyObject } from "./basic.js";
+import { Color } from "./color.js";
+import { Element } from "./element.js";
+import { ChartArea, Point } from "./geometric.js";
+import { LayoutItem, LayoutPosition } from "./layout.js";
+
+export { DateAdapter, TimeUnit, _adapters } from "./adapters.js";
+export {
+  Animation,
+  Animations,
+  Animator,
+  AnimationEvent,
+} from "./animation.js";
+export { Color } from "./color.js";
+export { Element } from "./element.js";
+export { ChartArea, Point } from "./geometric.js";
+export { LayoutItem, LayoutPosition } from "./layout.js";
 
 export interface ScriptableContext<TType extends ChartType> {
   active: boolean;
diff --git a/types/layout.d.ts b/types/layout.d.ts
index 4c770711582b035b9fca0abec301ac9296ec3a2d..12afd8d31d3d090f85792d24cc21faf8c1847af9 100644
--- a/types/layout.d.ts
+++ b/types/layout.d.ts
@@ -1,4 +1,4 @@
-import { ChartArea } from './geometric';
+import { ChartArea } from "./geometric.js";
 
 export type LayoutPosition = 'left' | 'top' | 'right' | 'bottom' | 'center' | 'chartArea' | {[scaleId: string]: number};
 